File: pec-observation-app/ios-app/Sources/PECObservation/Ports/NetworkClientPort.swift
import Foundation

public protocol NetworkClientPort {
    func get<T: Decodable>(path: String, token: String?) async throws -> T
    func post<T: Decodable>(path: String, body: Encodable, token: String?) async throws -> T
    func put(url: URL, data: Data, contentType: String) async throws
    func patch<T: Decodable>(path: String, body: Encodable, token: String?) async throws -> T
}

File: pec-observation-app/android-app/app/src/main/kotlin/br/gov/educacao/sp/pecobservation/ports/NetworkClientPort.kt
package br.gov.educacao.sp.pecobservation.ports

interface NetworkClientPort {
    suspend fun post(path: String, body: Map<String, Any>): Map<String, Any>
    suspend fun get(path: String): Map<String, Any>
    suspend fun patch(path: String, body: Map<String, Any>): Map<String, Any>
    suspend fun putBinary(url: String, filePath: String, contentType: String = "audio/mp4")
}
